services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,285 @@
+
+    def get_site_totals(self, version):
+        "Entrypoint for Flask routing for site totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.site_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return wrap_response(info)
+
+        return versioned_data
+
+    def get_table_totals(self, version):
+        "Entrypoint for Flask routing for table totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.table_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return wrap_response(info)
+
+        return versioned_data
+
+    def get_versions(self):
+        return list(self.field_totals.keys())
+
+
+def wrap_response(info):
+    content = json.dumps(info, indent=4, sort_keys=True)
+
+    resp = Response(content)
+
+    resp.headers['Content-Type'] = 'application/json'
+
+    return resp
+
+if __name__ == '__main__':
+    usage = """PEDSnet DQA Secondary Reports Service
+
+    Usage: main.py [--token=<token>] [--host=<host>] [--port=<port>] [--debug] [--interval=<interval>]
+
+    Options:
+        --help              Display the help.
+        --token=<token>     GitHub authorization token.
+        --host=<host>       Host of the service.
+        --port=<port>       Port of the service [default: 5000].
+        --debug             Enable debug output.
+        --interval=<interval>   Time interval, in minutes, to periodically check the repo for updates [default: 30]
+    """
+
+    from docopt import docopt
+
+    opts = docopt(usage)
+
+    host = opts['--host']
+    port = int(opts['--port'])
+    debug = opts['--debug']
+    interval = int(opts['--interval'])
+
+    # Set token if defined.
+    if opts['--token']:
+        GITHUB_AUTH_TOKEN = opts['--token']
+    else:
+        GITHUB_AUTH_TOKEN = os.environ.get('GITHUB_AUTH_TOKEN')
+
+    if not GITHUB_AUTH_TOKEN:
+        print('Authorization token required.')
+        sys.exit(1)
+
+    if debug:
+        logger.setLevel(logging.DEBUG)
+
+    dqa_resource = DQAResource(DIR_NAME)
+
+    # Do the initial fetch of the repository.
+    dqa_resource.update()
+
+    # Create the timer for repo update, and start it.
+    timer = ResourceTimer(dqa_resource, interval)
+    timer.start()
+
+    # Initialize the flask app and register the routes.
+    app = Flask(__name__)
+
+    CORS(app)
+
+    # Unique service ID and timestamp when it started.
+    SERVICE_ID = str(uuid.uuid4())
+    SERVICE_TIMESTAMP = datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
+
+    @app.route('/', methods=['GET'])
+    def index():
+        return json.dumps({
+            'name': 'DQA Service',
+            'version': __version__,
+            'time': SERVICE_TIMESTAMP,
+            'uuid': SERVICE_ID,
+        })
+
+    for version in dqa_resource.get_versions():
```

Comment:
Add `/pedsnet` that lists the available versions. This may not be used in the client, but provides a bit more visibility on what resources are available.